packages/lesswrong/server/repos/DialogueChecksRepo.ts

```diff
@@ -58,6 +63,32 @@ class DialogueChecksRepo extends AbstractRepo<"DialogueChecks"> {
       );
     `, [userId1, userId2])
   }
+
+  async getMatchFormYourTurn(): Promise<DialogueCheckWithExtraData[]> {
+    const result = await this.getRawDb().any(`
+      SELECT 
+        dc.*,
+        dmp_reciprocal._id AS "targetUserMatchPreferenceId"
+      FROM "DialogueChecks" AS dc
+      LEFT JOIN "DialogueMatchPreferences" AS dmp ON dc._id = dmp."dialogueCheckId"
+      LEFT JOIN
+          "DialogueChecks" AS dc_reciprocal
+          ON
+              dc."userId" = dc_reciprocal."targetUserId"
+              AND dc."targetUserId" = dc_reciprocal."userId"
+      LEFT JOIN "DialogueMatchPreferences" AS dmp_reciprocal ON dc_reciprocal._id = dmp_reciprocal."dialogueCheckId"
+      LEFT JOIN "Notifications" AS n ON n."extraData"->>'targetUserMatchPreferenceId' = dmp_reciprocal._id
```

Comment:
The formatting here is making it pretty difficult to read.  Generally prefer:
- `ON` on new lines after the `JOIN`
- the table & alias on the same line as the `JOIN`